apps/events/models.py

```diff
@@ -0,0 +1,109 @@
+from datetime import datetime, timedelta
+
+from django.db import models
+from pytz import UTC
+
+from apps.chat.models import ChatMessageType
+from apps.events.actions import log
+from apps.experiments.models import Experiment, ExperimentSession, SessionStatus
+from apps.utils.models import BaseModel
+
+ACTION_FUNCTIONS = {"log": log}
+
+
+class EventActionType(models.TextChoices):
+    LOG = "log"  # Prints the last message
+    SUMMARIZE = "summarize"  # requires prompt
+
+
+class EventAction(BaseModel):
+    action_type = models.CharField(choices=EventActionType.choices)
+    params = models.JSONField(null=True, blank=True)  # The parameters for the specific action
+
+
+class BaseTrigger(BaseModel):
+    action = models.OneToOneField(EventAction, on_delete=models.CASCADE, related_name="action")
+
+    class Meta:
+        abstract = True
+
+    def fire(self, session):
+        return ACTION_FUNCTIONS[self.action.action_type](session)
+
+
+class TimeoutTrigger(BaseTrigger):
+    experiment = models.ForeignKey(Experiment, on_delete=models.CASCADE, related_name="timeout_triggers")
+    delay = models.PositiveIntegerField(
+        help_text="The amount of time in seconds to fire this trigger.",
+    )
+    total_num_triggers = models.IntegerField(
+        default=1,
+        help_text="The number of times to fire this trigger",
+    )
+    end_conversation = models.BooleanField(
```

Comment:
Hmm, yes, maybe this warrants some discussion? 
For the `end_conversation` specifically - I agree that it is an "action", but it is also a side-effect of the final action occurring. I suppose this could be modeled with a separate Trigger and Action pair, where the trigger is that the final timeout has occurred.